client/src/components/EditInstanceView.tsx

```diff
@@ -5,70 +5,60 @@
 import {
   CourseData, CourseInstanceData, GradingScale, Period
 } from 'aalto-grades-common/types';
-import { Container, LinearProgress, Typography } from '@mui/material';
-import { useState, useEffect } from 'react';
+import { Form, Formik, FormikErrors, FormikTouched } from 'formik';
+import {
+  Box, Button, Container, LinearProgress, MenuItem, TextField, Typography
+} from '@mui/material';
+import { useState } from 'react';
 import { NavigateFunction, useNavigate, Params, useParams } from 'react-router-dom';
+import { UseQueryResult } from '@tanstack/react-query';
+import * as yup from 'yup';
 
-import EditInstanceForm from './edit-instance-view/EditInstanceForm';
 import AlertSnackbar from './alerts/AlertSnackbar';
+import UnsavedChangesDialog from './alerts/UnsavedChangesDialog';
 
-import { getCourse } from '../services/courses';
-import { createInstance, getSisuInstance } from '../services/instances';
-import { Message, State } from '../types';
+import {
+  useAddInstance, UseAddInstanceResult, useGetCourse, useGetSisuInstance
+} from '../hooks/useApi';
+import useSnackPackAlerts, { SnackPackAlertState } from '../hooks/useSnackPackAlerts';
+import { convertToClientGradingScale, formatSisuCourseType } from '../services/textFormat';
+import { State } from '../types';
 
 export default function EditInstanceView(): JSX.Element {
-  const { courseId, sisuInstanceId }: Params = useParams();
-
   const navigate: NavigateFunction = useNavigate();
+  const { courseId, sisuInstanceId }: Params = useParams();
 
-  const [instance, setInstance]: State<CourseInstanceData | null> =
-    useState<CourseInstanceData | null>(null);
-
-  const [alertOpen, setAlertOpen]: State<boolean> = useState(false);
-
-  const [messageInfo, setMessageInfo]: State<Message | null> =
-    useState<Message | null>(null);
+  if (!courseId)
+    return (<></>);
```

Comment:
Instead of returning blank pages in missing params, etc. I think we could use the React router [errorElement](https://reactrouter.com/en/main/route/error-element) and create a general use error page.

Reply:
New issue: #380